controllers/userController.js

```diff
@@ -133,7 +184,41 @@ const controller = {
     }
   },
 
-  deleteAccount: async (req, res) => {},
+  deleteAccount: async (req, res) => {
+    const profileOwner = await UserModel.findOne({ username: req.authUser.username });
+    const user = await UserModel.findOne({ username: req.params.username });
+    // authorisation check: whether user is the project owner
+    if (user?._id.toString() !== profileOwner?._id.toString()) {
+      return res.status(401).json({
+        error: 'User is not authorised to change this project',
+      });
+    }
+
+    try {
+      const hostProjects = await ProjectModel.find({ user_id: profileOwner?._id }, { _id: 1 });
+      if (hostProjects.length) {
+        for (let i = 0, len = hostProjects.length; i < len; i += 1) {
+          await ContributorModel.deleteMany({ project_id: hostProjects[i]._id });
```

Comment:
since we delete the Contributor, we will also need to delete all ContributorRelationships that have this contributor_id